### ISSUE TITLE
bug: additional properties on `property` type in JSON

### ISSUE THREAD

**Issue by agschrei** [CycloneDX/specification#371]
Currently the definition for a `property` expects keys `name` and `value` but neither of them is required and the presence of other keys is not prohibited through a constraint like `"additionalProperties": false`   

This is an inconsistency compared to other types defined on the schema such as `component` or `dependency` which disallow additional properties.

Concretely, that means this sample SBOM validates against the schema, but my argument is that it should not:

```json
{
  "$schema": "http://cyclonedx.org/schema/bom-1.4.schema.json",
  "bomFormat": "CycloneDX",
  "specVersion": "1.4",
  "version": 1,
  "metadata": {
    "component": {
      "bom-ref": "foo",
      "type": "application",
      "name": "foo",
      "properties": [
        {
          "name": "myCustomProperty",
          "value": "myCustomValue",
          "abc": ["def"]
        }
      ]
    }
  },
  "components": [
    {
      "bom-ref": "foo/bar",
      "type": "library",
      "name": "bar"
    }
  ]
}
```

https://www.jsonschemavalidator.net/s/dRHUBrfG

This seems like a small thing, but I just spent an hour trying to figure out why cyclonedx-core-java would validate an sbom just fine but then fail to parse it. So I think the spec would benefit from a stronger constraint on the `property` type.

I'm happy to contribute the spec changes for 1.6 if this gets approved.

**Comment by jkowalleck** [comment]
checked, can confirm for JSON: 
- name optional
- value optional
- additional properties allowed 
- see https://github.com/CycloneDX/specification/blob/8af880d5f2ba0a107de88a920a76cedd5ba75083/schema/bom-1.4.schema.json#L1147-L1162

in contrast to the JSON, the XML defines
- name is mandatory
- value is implicit kind of optional (implicit allows empty string)
- no additional properties/children allowed
- see https://github.com/CycloneDX/specification/blob/8af880d5f2ba0a107de88a920a76cedd5ba75083/schema/bom-1.4.xsd#L1721-L1734

in contrast to JSON, the Protobuff defines
- name is mandatory
- value is optional
- no additional properties/children allowed
- see https://github.com/CycloneDX/specification/blob/8af880d5f2ba0a107de88a920a76cedd5ba75083/schema/bom-1.5.proto#L606-L609

**Comment by stevespringett** [comment]
Yes please. Thanks @jkowalleck

**Comment by agschrei** [comment]
Thanks to both of you for the quick turnaround on this!

**Comment by stevespringett** [comment]
Thanks for reporting @agschrei . Fix will be included in the upcoming v1.6 specification.
